rpk reference: clarify that debug bundle must run inside pod (#1445)
redpanda-data/documentation: docs/reference/rpk/rpk-debug/shared/_bundle-contents-k8s.mdx

:::note

> rpk collects some data from the Kubernetes API. To do so, execute the rpk debug bundle command inside a Pod container that's running a Redpanda broker.

Reviewer: ```suggestion
rpk collects some of the bundle data from the Kubernetes API.
```

Reviewer: I added this detail to the top of the `rpk debug bundle` reference so that it is easier to find.
